docs/style-guidelines.md

```diff
@@ -1,20 +1,313 @@
+  * Use `[DisallowMultipleComponent]` attribute when there should be only one `Component` of such type on the `GameObject`
+* **Methods**
+  * Use `[PublicAPI]` attribute if `public method` is exposed to be called from **outside** of the solution 
+  * Use `[UsedImplicitly]` attribute for implicitly called methods, such as calls from Unity animation events, via Unity `GameObject.SendMessage()`, `GameObject.BroadcastMessage()` and similar. 
+  * Use `[Button]` attribute from `EasyButtons` (requires reference in `.asmdef` file) instead of `[ContectMenu]` attribute. Consider using `Mode` parameter of this attribute for `Editor`/`PlayMode`-only methods.
+* **Variables (in Inspector)**
+  * Use `[SerializedField]` for exposing variable in the inspector instead of converting it to public variable 
+  * Use CAPITAL letters for `[Header]` attribute 
+  * Use `[Space]` attribute for better grouping of exposed variables 
+  * Use `[Tooltip]` instead of comments for exposed in inspector variables
+
+### Scripting Symbols
+* try to avoid using Scripting Symbols. Often it can be solved via polymorphism 
+* when using it, try to decorate as less code as possible
+* No tabs when starting Scripting Symbols 
+* No line breaks between Scripting Symbols and its body
+ ```csharp 
+#if UNITY_WEBGL
+    // Here goes WEBGL code
+#elif UNITY_STANDALONE
+    // Here goes Standalone code
+#endif
+```
+### Comments
+* always use XML comments `/// <summary>`  before the `public` class declaration providing appropriate descriptions of their behavior
+ ```csharp 
+/// <summary>
+/// This is the InitialScene entry point.
+/// Most of the application subsystems should be initialized from this class Awake() event.
+/// </summary>
+public class Main: MonoBehaviour
+{
+    /// ...
+}
+```
+* **only** use comments `//` inside of the class when it is necessary and there is a need to provide additional information which cannot be covered by good naming:
+  * description of not obvious hidden logic behind the solution (such as for complex logic or mathematical algorithms) 
+  * reference to the bug-ticket
+  * "why" description of the hack
+  * XML comments `/// <summary>` for not obvious public methods 
+* each comment starts with an uppercase letter and end with a period
+* insert one space between the comment delimiter (`//`) and the comment text
+* don't use asterix syntax for comments 🔴 `/* comment */`
+* remove commented out code
+
+### Tests
+* **Class name** should contain Tests postfix after the feature name to the class that tests the current feature
+* **Test method name**
+  * try to reflect core of the **arrange**/**act**/**assert** part of the test in the test name (especially **assert**)
+  * consider using word `Should` to describe **assert** part and word `When` to describe **act** and **arrange** part of the test
+* The method body should be split by `// arrange` - `// act` - `// assert` comments to respective blocks. 
```

Comment:
This is just a silly thing that is the result of my manias but I always like to start all comments with upper case 😄

```
// Arrange

// Act

// Assert
```

Reply:
good one 💪 
There is also a possibility to make them CAPITAL to distinguish them from normal comments. However, since generally there are not so many comments in tests, I would go with first_capital as you proposed

Reply:
I completely agree!